### PR TITLE
Possibilty to override the 'Asset.timestamp' value in Configure when generating assetUrls

### PR DESCRIPTION
**Pull request by JoshuaLuckers**
See issue #11520

### CODE REVIEW

File: src/View/Helper/UrlHelper.php [hunk 1]
```diff
@@ -195,11 +218,14 @@ protected function _encodeUrl($url)
      * a timestamp will be added.
      *
      * @param string $path The file path to timestamp, the path must be inside WWW_ROOT
+     * @param bool|string $stamp If set will overrule the value of `Asset.timestamp` in Configure.
      * @return string Path with a timestamp added, or not.
      */
-    public function assetTimestamp($path)
+    public function assetTimestamp($path, $stamp = null)
```
Comment (ADmad) [inline src/View/Helper/UrlHelper.php]: We can't change the method arguments in a point/bugfix release.

Comment (ADmad) [reply src/View/Helper/UrlHelper.php]: You'll need to target your changes to `3.next` branch and they can be included in `3.6.0`.

Comment (JoshuaLuckers) [reply src/View/Helper/UrlHelper.php]: I guess I did something wrong. Is it better to close this pull request and make a new one?

Comment (markstory) [reply src/View/Helper/UrlHelper.php]: You can rebase your changes onto 3.next and the diff/commit list will be correct.

Comment (JoshuaLuckers) [reply src/View/Helper/UrlHelper.php]: Thanks for pointing me to the right direction.

File: src/View/Helper/UrlHelper.php [hunk 2]
```diff
@@ -71,6 +71,10 @@ public function build($url = null, $options = false)
      *   `fullBase` Return full URL with domain name
      *   `pathPrefix` Path prefix for relative URLs
      *   `plugin` False value will prevent parsing path as a plugin
+     *   `assetTimestamp` Overrides the value of `Asset.timestamp` in Configure.
```
Comment (ADmad) [inline src/View/Helper/UrlHelper.php]: Given that all methods are for assets perhaps a shorter name `timestamp` would be enough.

File: src/View/Helper/UrlHelper.php (hunk 1 above)
Comment (ADmad) [inline src/View/Helper/UrlHelper.php]: Change the argument name `$stamp` to `$timestamp` for consistency?

Comment (JoshuaLuckers) [reply src/View/Helper/UrlHelper.php]: I chose `$stamp` as argument name to keep it consistent with the variable used in the function.

Comment (dereuromark) [reply src/View/Helper/UrlHelper.php]: You can easily fix up all of them to "timestamp".

Comment (ADmad) [reply src/View/Helper/UrlHelper.php]: @JoshuaLuckers Internal variables don't show up in API docs but methods arguments and options do :)

File: src/View/Helper/UrlHelper.php [hunk 3]
```diff
@@ -195,12 +218,15 @@ protected function _encodeUrl($url)
...
      * @param string $path The file path to timestamp, the path must be inside WWW_ROOT
+     * @param bool|string $timestamp If set will overrule the value of `Asset.timestamp` in Configure.
      * @return string Path with a timestamp added, or not.
      */
-    public function assetTimestamp($path)
+    public function assetTimestamp($path, $timestamp = null)
     {
-        $stamp = Configure::read('Asset.timestamp');
-        $timestampEnabled = $stamp === 'force' || ($stamp === true && Configure::read('debug'));
+        if (is_null($timestamp)) {
```
Comment (markstory) [inline src/View/Helper/UrlHelper.php]: We generally use `$timestamp === null` as it doesn't use a function call.

File: tests/TestCase/View/Helper/UrlHelperTest.php [hunk 4]
```diff
@@ -254,6 +254,16 @@ public function testAssetUrlTimestampForce()
...
 
+    public function testAssetTimestampConfigureOverride()
+    {
+        $this->Helper->webroot = '';
+        Configure::write('Asset.timestamp', 'force');
+        $timestamp = false;
+
+        $result = $this->Helper->assetTimestamp(Configure::read('App.cssBaseUrl') . 'cake.generic.css', $timestamp);
+        $this->assertEquals(Configure::read('App.cssBaseUrl') . 'cake.generic.css', $result);
+    }
```
Comment (markstory) [inline tests/TestCase/View/Helper/UrlHelperTest.php]: We should also have tests for the new option in each of the public methods as that is new API surface area we're adding.

Comment (JoshuaLuckers) [reply tests/TestCase/View/Helper/UrlHelperTest.php]: Fair enough! I will add those later this week.

Comment (markstory) [reply tests/TestCase/View/Helper/UrlHelperTest.php]: Awesome!

**Pull request merged**